cub/cub/thread/thread_reduce.cuh

```diff
@@ -540,18 +539,38 @@ _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE AccumT ThreadReduce(const Input&
   using cub::internal::enable_simd_reduction;
   using cub::internal::enable_ternary_reduction;
   using PromT = ::cuda::std::_If<enable_promotion<Input, ReductionOp, AccumT>(), int, AccumT>;
+  _CCCL_IF_CONSTEXPR (!cub::detail::is_one_of<ReductionOp,
+                                              ::cuda::std::plus<>,
+                                              ::cuda::std::multiplies<>,
+                                              ::cuda::std::bit_and<>,
+                                              ::cuda::std::bit_or<>,
+                                              ::cuda::std::bit_xor<>,
+                                              ::cuda::maximum<>,
+                                              ::cuda::minimum<>,
+                                              cub::internal::SimdMin<ValueT>,
+                                              cub::internal::SimdMax<ValueT>>())
+  {
+    return cub::internal::ThreadReduceSequential<AccumT>(input, reduction_op);
+  }
+  _CCCL_IF_CONSTEXPR (cuda::std::is_same<ReductionOp, ::cuda::std::plus<>>::value
```

Comment:
Suggestion: Sadly, we may want to include more operators here:
```suggestion
  _CCCL_IF_CONSTEXPR (cuda::std::is_same<ReductionOp, ::cuda::std::plus<>>::value ||
                      cuda::std::is_same<ReductionOp, ::cuda::std::plus<ValueT>>::value
```
We may also just test whether ReductionOp is any instantiation of `::cuda::std::plus`.

Technically, this also does not include `thrust::plus`, but that will be fixed at the next major release.